shared/src/main/scala/pl/metastack/metatime/Pattern.scala

```diff
@@ -1,6 +1,6 @@
 package pl.metastack.metatime
 
-case class Pattern(parts: Seq[DateTime => String] = Nil) {
+case class Pattern(parts: Seq[Component => String] = Nil) {
```

Comment:
As you can see, some methods in `Pattern` are well-defined only for `DateTime` objects, while others expect `Time` objects. I would suggest to create add a type parameter to `Pattern` and make it a trait instead. Then, create two subclasses `DateTimePattern` and `TimePattern` which parameterise `Pattern` accordingly.